### ISSUE TITLE
Install failed on Ubuntu 16.04 with Ruby 2.4.1

### ISSUE THREAD

**Issue by IngoAlbers** [vmg/redcarpet#619]
I'm trying to install the gem with Ubuntu 16.04 using ruby 2.4.1 installed by rbenv.

```
ingo@ingo-XPS-13-9360:~$ gem install redcarpet --verbose
HEAD https://api.rubygems.org/api/v1/dependencies
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=redcarpet
200 OK
Getting SRV record failed: DNS result has no information for _rubygems._tcp.api.rubygems.org
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/COPYING
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/Gemfile
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/README.markdown
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/Rakefile
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/bin/redcarpet
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/autolink.c
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/autolink.h
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/buffer.c
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/buffer.h
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/extconf.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/houdini.h
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/houdini_href_e.c
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/houdini_html_e.c
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/html.c
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/html.h
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/html_blocks.h
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/html_smartypants.c
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/markdown.c
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/markdown.h
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/rc_markdown.c
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/rc_render.c
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/redcarpet.h
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/stack.c
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet/stack.h
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/lib/redcarpet.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/lib/redcarpet/cli.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/lib/redcarpet/compat.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/lib/redcarpet/render_man.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/lib/redcarpet/render_strip.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/redcarpet.gemspec
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/benchmark.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/custom_render_test.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/fixtures/benchmark.md
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/html5_test.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/html_render_test.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/html_toc_render_test.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/markdown_test.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/pathological_inputs_test.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/redcarpet_bin_test.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/redcarpet_compat_test.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/safe_render_test.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/smarty_html_test.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/smarty_pants_test.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/stripdown_render_test.rb
/home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/test/test_helper.rb
Building native extensions.  This could take a while...
current directory: /home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet
/home/ingo/.rbenv/versions/2.4.1/bin/ruby -r ./siteconf20170621-26093-nh63dp.rb extconf.rb
creating Makefile
current directory: /home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet
make "DESTDIR=" clean
current directory: /home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0/ext/redcarpet
make "DESTDIR="
compiling autolink.c
compiling buffer.c
buffer.c: In function ‘bufcstr’:
buffer.c:111:45: warning: passing argument 1 of ‘bufgrow’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  if (buf->size + 1 <= buf->asize || bufgrow(buf, buf->size + 1) == BUF_OK) {
                                             ^
buffer.c:60:1: note: expected ‘struct buf *’ but argument is of type ‘const struct buf *’
 bufgrow(struct buf *buf, size_t neosz)
 ^
buffer.c: At top level:
cc1: warning: unrecognized command line option ‘-Wno-self-assign’
cc1: warning: unrecognized command line option ‘-Wno-constant-logical-operand’
cc1: warning: unrecognized command line option ‘-Wno-parentheses-equality’
cc1: warning: unrecognized command line option ‘-Wno-tautological-compare’
compiling houdini_href_e.c
compiling houdini_html_e.c
compiling html.c
compiling html_smartypants.c
html_smartypants.c: In function ‘squote_len’:
html_smartypants.c:104:39: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  static char* single_quote_list[] = { "'", "&#39;", "&#x27;", "&apos;", NULL };
                                       ^
html_smartypants.c:104:44: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  static char* single_quote_list[] = { "'", "&#39;", "&#x27;", "&apos;", NULL };
                                            ^
html_smartypants.c:104:53: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  static char* single_quote_list[] = { "'", "&#39;", "&#x27;", "&apos;", NULL };
                                                     ^
html_smartypants.c:104:63: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  static char* single_quote_list[] = { "'", "&#39;", "&#x27;", "&apos;", NULL };
                                                               ^
html_smartypants.c: In function ‘smartypants_squote’:
html_smartypants.c:149:30: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
    uint8_t next_char = (size > 1+next_squote_len) ? text[1+next_squote_len] : 0;
                              ^
html_smartypants.c: In function ‘smartypants_cb__amp’:
html_smartypants.c:250:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  int len = squote_len(text, size);
  ^
html_smartypants.c: At top level:
cc1: warning: unrecognized command line option ‘-Wno-self-assign’
cc1: warning: unrecognized command line option ‘-Wno-constant-logical-operand’
cc1: warning: unrecognized command line option ‘-Wno-parentheses-equality’
cc1: warning: unrecognized command line option ‘-Wno-tautological-compare’
compiling markdown.c
markdown.c: In function ‘char_link’:
markdown.c:1045:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
   struct buf id = { 0, 0, 0, 0 };
   ^
markdown.c:1085:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
   size_t nb_p = 0;
   ^
markdown.c: In function ‘sd_markdown_render’:
markdown.c:2818:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  int footnotes_enabled  = md->ext_flags & MKDEXT_FOOTNOTES;
  ^
markdown.c:2836:43: warning: passing argument 1 of ‘is_codefence’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   if (codefences_enabled && (is_codefence(document + beg, doc_size - beg, NULL) != 0))
                                           ^
markdown.c:1389:1: note: expected ‘uint8_t * {aka unsigned char *}’ but argument is of type ‘const uint8_t * {aka const unsigned char *}’
 is_codefence(uint8_t *data, size_t size, struct buf *syntax)
 ^
markdown.c: At top level:
markdown.c:94:15: warning: ‘char_underline’ declared ‘static’ but never defined [-Wunused-function]
 static size_t char_underline(struct buf *ob, struct sd_markdown *rndr, uint8_t *data, size_t offset, size_t size);
               ^
cc1: warning: unrecognized command line option ‘-Wno-self-assign’
cc1: warning: unrecognized command line option ‘-Wno-constant-logical-operand’
cc1: warning: unrecognized command line option ‘-Wno-parentheses-equality’
cc1: warning: unrecognized command line option ‘-Wno-tautological-compare’
compiling rc_markdown.c
rc_markdown.c: In function ‘rb_redcarpet_md_render’:
rc_markdown.c:138:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  struct rb_redcarpet_rndr *renderer;
  ^
rc_markdown.c: At top level:
cc1: warning: unrecognized command line option ‘-Wno-self-assign’
cc1: warning: unrecognized command line option ‘-Wno-constant-logical-operand’
cc1: warning: unrecognized command line option ‘-Wno-parentheses-equality’
cc1: warning: unrecognized command line option ‘-Wno-tautological-compare’
compiling rc_render.c
rc_render.c: In function ‘rndr_link_attributes’:
rc_render.c:287:28: warning: variable ‘rndr’ set but not used [-Wunused-but-set-variable]
  struct rb_redcarpet_rndr *rndr;
                            ^
rc_render.c: At top level:
cc1: warning: unrecognized command line option ‘-Wno-self-assign’
cc1: warning: unrecognized command line option ‘-Wno-constant-logical-operand’
cc1: warning: unrecognized command line option ‘-Wno-parentheses-equality’
cc1: warning: unrecognized command line option ‘-Wno-tautological-compare’
compiling stack.c
linking shared-object redcarpet.so
redcarpet.so: final close failed: Invalid operation
collect2: error: ld returned 1 exit status
Makefile:257: recipe for target 'redcarpet.so' failed
make: *** [redcarpet.so] Error 1
ERROR:  Error installing redcarpet:
	ERROR: Failed to build gem native extension.

    Building has failed. See above output for more information on the failure.
make failed, exit code 2

Gem files will remain installed in /home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/redcarpet-3.4.0 for inspection.
Results logged to /home/ingo/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/extensions/x86_64-linux/2.4.0-static/redcarpet-3.4.0/gem_make.out
```

When I use ruby 2.3.0 though it works:

```
ingo@ingo-XPS-13-9360:~$ rbenv shell 2.3.0
ingo@ingo-XPS-13-9360:~$ gem install redcarpet
Building native extensions.  This could take a while...
Successfully installed redcarpet-3.4.0
Parsing documentation for redcarpet-3.4.0
Done installing documentation for redcarpet after 0 seconds
1 gem installed
```

**Comment by robin850** [comment]
Hi @IngoAlbers,

Actually I don't think that's a problem with Redcarpet. The code is tested against 2.4 on Travis CI so this must be a problem with your setup. I may be wrong but I guess that's because, by default, Ubuntu provides Ruby 2.3 so the header files for this version are present but the headers for 2.4 aren't.

Could you also paste the result of `gem install redcarpet --verbose` running against Ruby 2.3 please ?

**Comment by IngoAlbers** [comment]
Hi @robin850, 
thanks for the feedback and trying to help even though it should be my specific setup. 
Here is the result verbose log using ruby 2.3:

```
ingo@ingo-XPS-13-9360:~$ rbenv shell 2.3.0
ingo@ingo-XPS-13-9360:~$ gem install redcarpet --verbose
HEAD https://api.rubygems.org/api/v1/dependencies
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=redcarpet
200 OK
Getting SRV record failed: DNS result has no information for _rubygems._tcp.api.rubygems.org
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/COPYING
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/Gemfile
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/README.markdown
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/Rakefile
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/bin/redcarpet
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/autolink.c
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/autolink.h
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/buffer.c
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/buffer.h
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/extconf.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/houdini.h
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/houdini_href_e.c
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/houdini_html_e.c
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/html.c
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/html.h
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/html_blocks.h
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/html_smartypants.c
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/markdown.c
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/markdown.h
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/rc_markdown.c
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/rc_render.c
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/redcarpet.h
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/stack.c
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet/stack.h
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/lib/redcarpet.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/lib/redcarpet/cli.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/lib/redcarpet/compat.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/lib/redcarpet/render_man.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/lib/redcarpet/render_strip.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/redcarpet.gemspec
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/benchmark.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/custom_render_test.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/fixtures/benchmark.md
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/html5_test.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/html_render_test.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/html_toc_render_test.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/markdown_test.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/pathological_inputs_test.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/redcarpet_bin_test.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/redcarpet_compat_test.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/safe_render_test.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/smarty_html_test.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/smarty_pants_test.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/stripdown_render_test.rb
/home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/test/test_helper.rb
Building native extensions.  This could take a while...
current directory: /home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet
/home/ingo/.rbenv/versions/2.3.0/bin/ruby -r ./siteconf20170621-1901-y1i912.rb extconf.rb
creating Makefile
current directory: /home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet
make "DESTDIR=" clean
current directory: /home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet
make "DESTDIR="
compiling markdown.c
markdown.c: In function ‘char_link’:
markdown.c:1045:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
   struct buf id = { 0, 0, 0, 0 };
   ^
markdown.c:1085:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
   size_t nb_p = 0;
   ^
markdown.c: In function ‘sd_markdown_render’:
markdown.c:2818:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  int footnotes_enabled  = md->ext_flags & MKDEXT_FOOTNOTES;
  ^
markdown.c:2836:43: warning: passing argument 1 of ‘is_codefence’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   if (codefences_enabled && (is_codefence(document + beg, doc_size - beg, NULL)
                                           ^
markdown.c:1389:1: note: expected ‘uint8_t * {aka unsigned char *}’ but argument is of type ‘const uint8_t * {aka const unsigned char *}’
 is_codefence(uint8_t *data, size_t size, struct buf *syntax)
 ^
markdown.c: At top level:
markdown.c:94:15: warning: ‘char_underline’ declared ‘static’ but never defined [-Wunused-function]
 static size_t char_underline(struct buf *ob, struct sd_markdown *rndr, uint8_t 
               ^
compiling rc_render.c
rc_render.c: In function ‘rndr_link_attributes’:
rc_render.c:287:28: warning: variable ‘rndr’ set but not used [-Wunused-but-set-variable]
  struct rb_redcarpet_rndr *rndr;
                            ^
compiling houdini_href_e.c
compiling html.c
compiling rc_markdown.c
rc_markdown.c: In function ‘rb_redcarpet_md_render’:
rc_markdown.c:138:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  struct rb_redcarpet_rndr *renderer;
  ^
compiling stack.c
compiling autolink.c
compiling html_smartypants.c
html_smartypants.c: In function ‘squote_len’:
html_smartypants.c:104:39: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  static char* single_quote_list[] = { "'", "&#39;", "&#x27;", "&apos;", NULL };
                                       ^
html_smartypants.c:104:44: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  static char* single_quote_list[] = { "'", "&#39;", "&#x27;", "&apos;", NULL };
                                            ^
html_smartypants.c:104:53: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  static char* single_quote_list[] = { "'", "&#39;", "&#x27;", "&apos;", NULL };
                                                     ^
html_smartypants.c:104:63: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  static char* single_quote_list[] = { "'", "&#39;", "&#x27;", "&apos;", NULL };
                                                               ^
html_smartypants.c: In function ‘smartypants_squote’:
html_smartypants.c:149:30: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
    uint8_t next_char = (size > 1+next_squote_len) ? text[1+next_squote_len] : 0
                              ^
html_smartypants.c: In function ‘smartypants_cb__amp’:
html_smartypants.c:250:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  int len = squote_len(text, size);
  ^
compiling buffer.c
buffer.c: In function ‘bufcstr’:
buffer.c:111:45: warning: passing argument 1 of ‘bufgrow’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  if (buf->size + 1 <= buf->asize || bufgrow(buf, buf->size + 1) == BUF_OK) {
                                             ^
buffer.c:60:1: note: expected ‘struct buf *’ but argument is of type ‘const struct buf *’
 bufgrow(struct buf *buf, size_t neosz)
 ^
compiling houdini_html_e.c
linking shared-object redcarpet.so
current directory: /home/ingo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/redcarpet-3.4.0/ext/redcarpet
make "DESTDIR=" install
/usr/bin/install -c -m 0755 redcarpet.so ./.gem.20170621-1901-1j0i30s

/home/ingo/.rbenv/versions/2.3.0/bin/redcarpet
Successfully installed redcarpet-3.4.0
Parsing documentation for redcarpet-3.4.0
Parsing sources...
100% [38/38]  lib/redcarpet/render_strip.rb   
Done installing documentation for redcarpet after 0 seconds
1 gem installed
```

**Comment by robin850** [comment]
Hum, this is interesting. I don't know if you changed something on your setup but the output isn't the same between Ruby 2.3.0 and 2.4.1. It looks like you're using another compiler as warnings like:

~~~
cc1: warning: unrecognized command line option ‘-Wno-self-assign’
cc1: warning: unrecognized command line option ‘-Wno-constant-logical-operand’
cc1: warning: unrecognized command line option ‘-Wno-parentheses-equality’
cc1: warning: unrecognized command line option ‘-Wno-tautological-compare’
~~~

aren't present compiling under 2.3 but are compiling with 2.4. Did you change something on your machine since you created this ticket ?

**Comment by IngoAlbers** [comment]
I just tested it again using `rbenv shell 2.4.1` and `rbenv shell 2.3.0` and can confirm that both logs that I posted are still the ones I get.

You are right about the warnings though, which confused me too. Not sure if it might be related to rbenv. Really strange. Might be that I did something wrong at some point when setting up my new development environment. I might start from a clean system again tomorrow.
